py/mpprint.h

```diff
@@ -52,6 +52,14 @@ typedef struct _mp_print_t {
     mp_print_strn_t print_strn;
 } mp_print_t;
 
+typedef struct _mp_print_ext_t {
+    mp_print_t base;
+    const char* item_separator;
```

Comment:
I'm not sure, but it might be possible to save code size (at the expense of some performance) by using qstr instead, and then later on `mp_vprintf` to print it. Seems worth trying anyway just to check.

Reply:
I am not sure how to do this with `qstr`'s but I am open to ideas.
I think `qstr` with spaces and commas aren't possible in C ?

Reply:
> I am not sure how to do this with qstr's but I am open to ideas.

Untested but I assume something like: use `qstr` as member in the struct, then use `mp_obj_str_get_qstr` to get ift from the arguments, then `mp_printf(print, "%q", separator)` to print.

> I think qstr with spaces and commas aren't possible in C ?

Indeed, but they can be added in makeqtrdata.py

Reply:
Wouldn't this change the default in list/dict/tuple-print to always require a `mp_printf` instead of just `mp_print_str` ?

this:
```
const char *item_separator = ", ";
...
mp_print_str(print, item_separator);
```

to this:
```
qstr item_separator = MP_ROM_QSTR(MP_QSTR_: ); 
//...
mp_printf(print, "%q", item_separator);
```

I can only see how this would affect very long separators, since `mp_print_str` always uses `strlen`.